### PR TITLE
Port Planning scene monitor

### PR DESCRIPTION
**Pull request by anasarrak**
Needs further work, but for now lets merge it

### CODE REVIEW

File: moveit_ros/planning/planning_scene_monitor/demos/demo_scene.cpp [hunk 1]
```diff
@@ -75,23 +84,27 @@ void sendKnife()
...
-  spinner.start();
+  rclcpp::init(argc, argv);
+  rclcpp::executors::SingleThreadedExecutor executor;
 
-  sendKnife();
+  node = rclcpp::Node::make_shared("demo");
 
-  ros::waitForShutdown();
+  while (!shutdown_req){
+  sendKnife();
```
Comment (vmayoral) [inline moveit_ros/planning/planning_scene_monitor/demos/demo_scene.cpp]: Why's this not tabbed appropriately @anasarrak? Please have a look at https://moveit.ros.org/documentation/contributing/code/. Particularly, `clang-format Auto Code Formatting`. While I'm not sure what's their policy for accepting changes here (they complained a few times when I runned it as they requested), I think it's a good practice we should also apply.

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h [hunk 2]
```diff
@@ -462,13 +464,14 @@ class PlanningSceneMonitor : private boost::noncopyable
...
-  ros::Time last_update_time_;                     /// Last time the state was updated
-  ros::Time last_robot_motion_time_;               /// Last time the robot has moved
+  rclcpp::Time last_update_time_;                     /// Last time the state was updated
+  rclcpp::Time last_robot_motion_time_;               /// Last time the robot has moved
 
-  ros::NodeHandle nh_;
-  ros::NodeHandle root_nh_;
-  ros::CallbackQueue queue_;
-  std::shared_ptr<ros::AsyncSpinner> spinner_;
+  // ros::NodeHandle nh_;
```
Comment (vmayoral) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h]: If we're not using these handles, maybe we can remove them?

**Comment by vmayoral** [comment]
Code fails in OS X @anasarrak:

```bash
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp:160:81: error: cannot pass object of non-trivial type 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') through variadic function; call will abort at runtime [-Wnon-pod-varargs]
    RCLCPP_DEBUG(node_->get_logger(),"Listening to joint states on topic '%s'", joint_states_topic);
                                                                                ^
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp:252:43: error: cannot pass object of non-trivial type 'rclcpp::Duration' through variadic function; call will abort at runtime [-Wnon-pod-varargs]
                joint->getName().c_str(), (now - it->second) , age.seconds());
                                          ^
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp:282:43: error: cannot pass object of non-trivial type 'rclcpp::Duration' through variadic function; call will abort at runtime [-Wnon-pod-varargs]
                joint->getName().c_str(), (now - it->second) , age.seconds());
                                          ^
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp:465:70: error: cannot pass object of non-trivial type 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') through variadic function; call will abort at runtime [-Wnon-pod-varargs]
                              "and '%s' with TF exception: ", joint->getName(), parent_frame.c_str(), child_frame.c_str(), ex.what());
                                                                     ^
```

**Comment by vmayoral** [comment]
Probably an argument missing `c_str()`. But it's just a guess.

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h (hunk 2 above)
Comment (anasarrak) [reply moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h]: It was under the private: part, I switched it to protected as the nh_

https://github.com/ros-planning/moveit2/commit/5e641fde06500336ec31a933092439359628c41e

**Comment by anasarrak** [comment]
> Probably an argument missing `c_str()`. But it's just a guess.

This normally happens with the const strings, for some reason linux is more permisive.

Fixed : https://github.com/AcutronicRobotics/moveit2/pull/39/commits/5577fd410c38c9f3e521dc502b6868edcec76143

File: moveit_ros/planning/planning_scene_monitor/demos/demo_scene.cpp (hunk 1 above)
Comment (anasarrak) [reply moveit_ros/planning/planning_scene_monitor/demos/demo_scene.cpp]: Fine, I used the following command to format just a single file
```
find -name demo_scene.cpp | xargs clang-format-3.9 -i -style=file $1
```
The result: https://github.com/AcutronicRobotics/moveit2/pull/39/commits/6eb5254eaed0b35328dd982907afe2825cf777e5

**Comment by vmayoral** [comment]
More issues with OS X, I think I'm going to re-enable Travis for OS X:

```bash
...
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp:252:43: error: cannot pass object of non-trivial type 'rclcpp::Duration' through variadic function; call will abort at runtime [-Wnon-pod-varargs]
                joint->getName().c_str(), (now - it->second) , age.seconds());
                                          ^
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp:282:43: error: cannot pass object of non-trivial type 'rclcpp::Duration' through variadic function; call will abort at runtime [-Wnon-pod-varargs]
                joint->getName().c_str(), (now - it->second) , age.seconds());
                                          ^
```

**Comment by vmayoral** [comment]
CI passed. Let's merge it (note: @anasarrak, the issues found above clearly only show in OS X (when built from source?)). I'll re-enable OS X in Travis, please take it into account for future PRs and ensure it passes in both please.

**Pull request merged**